src/Microsoft.VisualStudio.Threading/AsyncLazy.cs

```diff
@@ -196,6 +193,47 @@ public Task<T> GetValueAsync(CancellationToken cancellationToken)
             return this.value.WithCancellation(cancellationToken);
         }
 
+        /// <summary>
+        /// Gets the lazily computed value.
+        /// </summary>
+        /// <returns>The lazily constructed value.</returns>
+        /// <exception cref="InvalidOperationException">
+        /// Thrown when the value factory calls <see cref="GetValueAsync()"/> on this instance.
+        /// </exception>
+        public T GetValue() => this.GetValue(CancellationToken.None);
+
+        /// <summary>
+        /// Gets the lazily computed value.
+        /// </summary>
+        /// <param name="cancellationToken">
+        /// A token whose cancellation indicates that the caller no longer is interested in the result.
+        /// Note that this will not cancel the value factory (since other callers may exist).
+        /// But when this token is canceled, the caller will experience an <see cref="OperationCanceledException"/>
+        /// immediately and a dis-joining of any <see cref="JoinableTask"/> that may have occurred as a result of this call.
+        /// </param>
+        /// <returns>The lazily constructed value.</returns>
+        /// <exception cref="InvalidOperationException">
+        /// Thrown when the value factory calls <see cref="GetValueAsync()"/> on this instance.
+        /// </exception>
+        /// <exception cref="OperationCanceledException">Thrown when <paramref name="cancellationToken"/> is canceled before the value is computed.</exception>
+        public T GetValue(CancellationToken cancellationToken)
```

Comment:
Those methods are fine.  However, will this change make developers more likely to use those methods, instead of the async version, even under an async method chain and may not see a JTF.Run inside it?  Will this miss usage be caught by validation rules?  Or maybe we should add EditBrowseable to make it less discoverable?

Reply:
Your concerns are exactly why it's taken me this long to finally add this overload. But with vs-threading 15.8 our package depends on the analyzer package, ensuring that everyone using nuget to acquire the vs-threading library also gets the [VSTHRD103](https://github.com/Microsoft/vs-threading/blob/master/doc/analyzers/VSTHRD103.md) rule which will flag any use of `GetValue` from an async method and encourage them to use `GetValueAsync` instead. 
In an effort to fix deadlocks, the [VSTHRD011](https://github.com/Microsoft/vs-threading/blob/master/doc/analyzers/VSTHRD011.md) is getting more aggressive in getting people to switch from `Lazy<T>` to `AsyncLazy<T>` and we need to make it more tolerable for people to switch given much of their code still runs synchronously. 

Because of the pressing need to make transition easier to fix VSTHRD011 issues, and the VSTHRD103 rule helping people stay async when they can be, it seems now is the right time to add this method.

Reply:
I noticed that the solution of 011 rule asks the developer to switch to AsyncLazy, but didn't mention that they also need pass in a JTF to the constructor.  Should we recommend that as well?

Reply:
We do: encouraging users to supply a JTF instance parameter is the job of [VSTHRD012](https://github.com/Microsoft/vs-threading/blob/master/doc/analyzers/VSTHRD012.md). :smile:

Reply:
Aha, thanks, only pay attention to a single rule.